components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushPatientDataTest.java

```diff
@@ -0,0 +1,154 @@
+        Assert.assertEquals(method.invoke(this.mocker.getComponentUnderTest(), serverConfiguration),
+                "http://apples/bin/receivePatientData");
+    }
+
+    @Test
+    public void generateRequestDataTest() throws ComponentLookupException, NoSuchMethodException, IllegalAccessException,
+            InvocationTargetException {
+        // set up access for the private method
+        Method method = DefaultPushPatientData.class.getDeclaredMethod("generateRequestData",
+                String.class, String.class,
+                String.class, String.class);
+        method.setAccessible(true);
+
+        List<NameValuePair> result = (List<NameValuePair>) method.invoke(this.mocker.getComponentUnderTest(),
+                "actionName", "userName",
+                "passWord", "userToken");
+        Assert.assertEquals("xpage", result.get(0).getName());
+        Assert.assertEquals("plain", result.get(0).getValue());
+        Assert.assertEquals("push_protocol_version", result.get(1).getName());
+        Assert.assertEquals("1", result.get(1).getValue());
+        Assert.assertEquals("action", result.get(2).getName());
+        Assert.assertEquals("actionName", result.get(2).getValue());
+        Assert.assertEquals("username", result.get(3).getName());
+        Assert.assertEquals("userName", result.get(3).getValue());
+        Assert.assertEquals("user_login_token", result.get(4).getName());
+        Assert.assertEquals("userToken", result.get(4).getValue());
+    }
+
+    @Test
+    public void generateRequestDataNoTokenTest() throws ComponentLookupException, NoSuchMethodException, IllegalAccessException,
+            InvocationTargetException
+    {
+        // set up access for the private method
+        Method method = DefaultPushPatientData.class.getDeclaredMethod("generateRequestData",
+                String.class, String.class,
+                String.class, String.class);
+        method.setAccessible(true);
+
+        // check that the last
+        List<NameValuePair> result = (List<NameValuePair>) method.invoke(this.mocker.getComponentUnderTest(),
+                "actionName", "userName",
+                "passWord", "");
+        Assert.assertEquals("password", result.get(4).getName());
+        Assert.assertEquals("passWord", result.get(4).getValue());
+    }
+
+    @Test
+    public void getRemoteConfigurationPerformsCorrectly() throws ComponentLookupException, XWikiException
+    {
+        when(this.execution.getContext()).thenReturn(this.executionContext);
+        when(this.executionContext.getProperty(Matchers.anyString())).thenReturn(this.xWikiContext);
+        when(this.xWikiContext.getWiki()).thenReturn(xWiki);
+        when(xWiki.getDocument(Matchers.any(DocumentReference.class), Matchers.any(XWikiContext.class))).thenReturn(this.xWikiDocument);
+        this.mocker.getComponentUnderTest().getRemoteConfiguration("id", "name", "pass", "token");
```

Comment:
This fails because you're not mocking `context.getDatabase()` (which is deprecated and should be replaced with `getWikiId()`).